src/arrayMethodJoin.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable use-isnan */
+/* eslint-disable no-console */
 'use strict';
 
 /**
  * Implement method join
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const sep = String(separator);
+
+    for (let i = 0; i < this.length; i++) {
+      switch (true) {
+        case (i + 1) !== this.length && this[i] === undefined:
+        case (i + 1) !== this.length && this[i] === null:
+        case (i + 1) !== this.length && this[i] === NaN:
+          result += ',';
+          break;
+        case this[i] === undefined:
+          break;
+        case (i + 1) !== this.length && separator === undefined:
```

Comment:
no need to check `separator === undefined` - just set the default value in function arguments